Modules/_winapi.c

```diff
@@ -2192,8 +2176,10 @@ static int winapi_exec(PyObject *m)
     WINAPI_CONSTANT(F_DWORD, SEC_NOCACHE);
     WINAPI_CONSTANT(F_DWORD, SEC_RESERVE);
     WINAPI_CONSTANT(F_DWORD, SEC_WRITECOMBINE);
+#ifndef MS_WINDOWS_NON_DESKTOP
     WINAPI_CONSTANT(F_DWORD, STARTF_USESHOWWINDOW);
     WINAPI_CONSTANT(F_DWORD, STARTF_USESTDHANDLES);
```

Comment:
This is another peculiar limitation in "winbase.h". They allow `CreateProcessW()` in "processthreadsapi.h" for the app, system, and games partitions, and it's the same for the process creation flags (e.g. `CREATE_BREAKAWAY_FROM_JOB`) and `STARTUPINFO[EX]W` record in "winbase.h", as well as `GetStdHandle()` and `SetStdHandle()` in "processenv.h". Then they throw a wrench in the works. All of the `STARTF_*` flags in "winbase.h" are limited to the desktop API.

The `subprocess` module requires the above two flags. Try to manually define them and test whether they work. Maybe `STARTF_USESHOWWINDOW` is irrelevant on Xbox. That's not a showstopper. I would hope that `STARTF_USESTDHANDLES` works. If it doesn't, `subprocess` should raise an informative exception if `stdin`, `stdout`, or `stderr` is used on Xbox.

```c
// winbase.h limits the STARTF_* flags to the desktop API as of 10.0.19041.
#ifndef STARTF_USESHOWWINDOW
#define STARTF_USESHOWWINDOW 0x00000001
#endif

#ifndef STARTF_USESTDHANDLES
#define STARTF_USESTDHANDLES 0x00000100
#endif
```

It would be nice to have a cheap way to check for a supported API partition. Maybe add `sys.winapi_partition`, with boolean attributes `desktop`, `app` (all app families), `pc_app` (desktop and PC app families), `phone_app`, `games` (desktop and games app families), `system` (system and server app families), and `server`. 

For example, check for support via `sys.winapi_partion.desktop or sys.winapi_partition.system`. 

For a regular desktop build, the attributes `desktop`, `app`, `pc_app`, and `games` would all be true. For Xbox, only `games` would be true.

Reply:
I added them for testing purposes using:
```
    import subprocess
    print(subprocess.run(["dir"], shell=True, capture_output=True))
    subprocess.run("exit 1", shell=True, check=True)
```

This did run without problems, so I will add them in.

I will look into exposing the `winapi_partition` via the `sys` module.

Reply:
let's add something like `sys.winapi_partition` after the current PRs are done. The exact information exposed should probably be discussed and this PR is already pretty cluttered.